modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/GroupResourceManagerTest.java

```diff
@@ -0,0 +1,881 @@
+    }
+
+    @AfterMethod
+    public void tearDown() {
+
+    }
+
+    private SCIMResponse getEncodeSCIMExceptionObject(AbstractCharonException exception) {
+
+        JSONEncoder encoder = new JSONEncoder();
+        Map<String, String> responseHeaders = new HashMap<>();
+        responseHeaders.put(SCIMConstants.CONTENT_TYPE_HEADER, SCIMConstants.APPLICATION_JSON);
+        return new SCIMResponse(exception.getStatus(), encoder.encodeSCIMException(exception), responseHeaders);
+    }
+
+    private Group getNewGroup() throws BadRequestException, CharonException, InternalErrorException {
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getGroupResourceSchema();
+        JSONDecoder decoder = new JSONDecoder();
+
+        return decoder.decodeResource(NEW_GROUP_SCIM_OBJECT_STRING, schema, new Group());
+    }
+
+    @DataProvider(name = "dataForGetGroupSuccess")
+    public Object[][] dataToGetGroupSuccess() throws CharonException, InternalErrorException, BadRequestException {
+
+        Group group = getNewGroup();
+        String id = group.getId();
+
+        return new Object[][]{
+                {id, null, null, group},
+                {id, null, "members", group},
+                {id, "displayName", null, group},
+                {id, "displayName", "members", group}
+        };
+    }
+
+    @Test(dataProvider = "dataForGetGroupSuccess")
+    public void testGetGroupSuccess(String id, String attributes, String excludeAttributes, Object objectGroup)
+            throws BadRequestException, NotFoundException, CharonException, NotImplementedException {
+
+        Group group = (Group) objectGroup;
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getGroupResourceSchema();
+        Map<String, Boolean> requiredAttributes = ResourceManagerUtil.getOnlyRequiredAttributesURIs(
+                (SCIMResourceTypeSchema) CopyUtil.deepCopy(schema), attributes, excludeAttributes);
+
+        mockStatic(AbstractResourceManager.class);
+
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_GROUP_ENDPOINT);
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+        when(userManager.getGroup(id, requiredAttributes)).thenReturn(group);
+        SCIMResponse scimResponse = groupResourceManager.get(id, userManager, attributes, excludeAttributes);
+        JSONObject obj = new JSONObject(scimResponse.getResponseMessage());
+        if (attributes != null) {
+            Assert.assertTrue(obj.has(attributes));
+        }
+        if (excludeAttributes != null) {
+            Assert.assertFalse(obj.has(excludeAttributes));
+        }
+        Assert.assertEquals(scimResponse.getResponseStatus(), ResponseCodeConstants.CODE_OK);
+    }
+
+    @DataProvider(name = "dataForGetGroupNotExceptions")
```

Comment:
```suggestion
    @DataProvider(name = "dataForGetGroupNotFoundExceptions")
```

Reply:
Its not the data provider for ``dataForGetGroupNotFoundExceptions`` test only, its common for all the exceptions in get method. Typo Not is removed in commit 1c9cf8aca69b6c33157e750da50f23315f9947fc